compat/src/index.d.ts

```diff
@@ -18,7 +18,6 @@ declare namespace React {
 	export import PropRef = _hooks.PropRef;
 	export import Reducer = _hooks.Reducer;
 	export import Dispatch = _hooks.Dispatch;
-	export import Ref = _hooks.Ref;
```

Comment:
It's ok if you want to call it out-of-scope for this PR, but should `_hooks.Ref` even exist? Should hooks also pull the `Ref` type from core?

Reply:
The type that corresponds to `_hooks.Ref` is called [`RefObject` in React](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/6c7875bb2df2e55876097b8aa15c2aa6644641f0/types/react/index.d.ts#L150).

preact also defines `RefObject` type in core, which is similar to both `_hooks.Ref` and React's `RefObject`. So I can assume that hooks should pull `RefObject` type from core.

Reply:
Yes, that was what I was asking. Seems like the root issue is a diverging type in `/hooks` rather than anything here.

Reply:
So is there anything left to be done in this pull request, or is it good to go?

Reply:
Well ideally we'd fix & unify all ref types, as I'd guess `PropRef` might cause similar problems, but if you don't want to fix it I suppose it can be done separately

Reply:
No problem, I will look deeper into how we can unify these types. I just wanted to clarify the scope of this issue.

Reply:
> Well ideally we'd fix & unify all ref types

I have tried to address the usage of a `Ref` type in hooks. Please, check.

> I'd guess `PropRef` might cause similar problems

`PropRef` is marked as deprecated, and there are no usages of it, except in one test.

Reply:
> PropRef is marked as deprecated, and there are no usages of it, except in one test.

Moreso that it's a rebranding of `MutableRef`, which, to my knowledge, shouldn't really exist either?

But whatever, doesn't matter much.